### ISSUE TITLE
Undesirable grid element focus ring that cannot be disabled.

### ISSUE THREAD

**Issue by jflamy-dev** [vaadin/flow#17521]
### Description of the bug

Clicking in a grid in SINGLE selection mode to open a dialog.  When exiting the dialog, the cell that was clicked to open the dialog shows a focus ring.  

This is a new v24 behavior: Vaadin v14 and v23 there is **no** focus ring on the selected cell.
Adding a cell focus listener indicates that the focus event is from the client, and not from the server itself.

See below for a sequence that reproduces the issue.

### Expected behavior

The documentation for Java Usage for v24 Grid says that focus ring is normally triggered by keyboard navigation.  There should be a way to disable this focus, or an easy way to clear it programmatically from inside the cell focus listener itself.

### Minimal reproducible example

Initial grid, with no selection.  The grid is is in single selection mode, with a listener that opens a dialog.
![image](https://github.com/vaadin/flow/assets/53801001/4d01a77a-ac9b-4868-9589-7dd48ae5b2cb)

Clicked on the M2 cell of the second row.  Notice there is no highlight on the cell.
![image](https://github.com/vaadin/flow/assets/53801001/08b258c5-acbf-4cee-aec0-3d7427b87053)

Immediately exit the dialog by typing ESC. The cell clicked to select the row has focus.  This focus is undesirable and confusing to the user.
![image](https://github.com/vaadin/flow/assets/53801001/561b2335-8bd9-4c4a-a3cd-020ae89e9a4d)

### Versions

Vaadin: 24.1.4
Flow: 24.1.5
Java: JetBrains s.r.o. 17.0.6
OS: amd64 Windows 11 10.0
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.62
Live reload: Java unavailable (HotswapAgent): Front end unavailable
